[Bug]: z.request when raw is true does not include buffer method on response object

### Bug Description

I'm not sure if this is an issue with your TypeScript type definitions or with the actual library, but contrary to the examples and error messages you receive, there is no `buffer()` method on `RawHttpResponse` or `BaseHttpResponse`.

[The following example is invalid, at least for TS](https://platform.zapier.com/reference/cli-docs#http-response-object):
![image](https://github.com/zapier/zapier-platform/assets/83599748/ae965788-c875-4e2b-98ee-c6e6c46020f5)


### Reproduction Steps

```typescript
import { ZObject } from 'zapier-platform-core' // 15.5.1 per package.json

const file = await z.request({
    url: download.item,
    raw: true,
})

// src/Hydrators.ts(34,44): error TS2339: Property 'buffer' does not exist on type 'RawHttpResponse'.
const buffer = await file.buffer() // CHOKE
```

### Zapier Platform version

15.5.1

### Node.js version

18.19.0

### Your Operating System

macOS 14.2.1

### npm/yarn version

10.2.3

### App ID

198292

### More Details

_No response_

--- a/packages/core/index.d.ts
+++ b/packages/core/index.d.ts
@@ -115,9 +115,10 @@ export interface HttpResponse extends BaseHttpResponse {
 }
 
 export interface RawHttpResponse extends BaseHttpResponse {
-  content: Buffer;
-  json: Promise<object | undefined>;
   body: NodeJS.ReadableStream;
+	buffer(): Promise<Buffer>;
+	json(): Promise<object>;
+	text(): Promise<string>;
 }
 
 type DehydrateFunc = <T>(

